serde_derive/src/internals/ctxt.rs

```diff
@@ -10,18 +10,27 @@ use std::cell::RefCell;
 use std::fmt::Display;
 use std::thread;
 
+/// A type to collect errors together and format them.
+///
+/// Dropping this object will cause a panic. It must be consumed using `check`.
+///
+/// References can be shared since this type uses run-time exclusive mut checking.
 #[derive(Default)]
 pub struct Ctxt {
+    // The contents will be set to `None` during checking. This is so that checking can be
+    // enforced.
     errors: RefCell<Option<Vec<String>>>,
 }
 
 impl Ctxt {
+    /// Create a new context object
```

Comment:
I don't think that explaining `fn new` creates a new object adds any value. Please remove this comment.

Reply:
I think I included this doc because the `Drop` behavior is non-standard. I've expanded it a little. If you still don't think it's necessary I'm happy to remove :).